resp/resp3/resp.go

```diff
@@ -158,6 +158,15 @@ func discardAttribute(br resp.BufferedReader, o *resp.Opts) error {
 	return discardMulti(br, attrHead.NumPairs*2, o)
 }
 
+// HasPrefix returns true if the next value in the given reader has the given
```

Comment:
I'm somewhat uneasy about this not returning the error. I guess in most cases the error will probably get returned by the next read operation which is performed, but I think there's still opportunity for weird edge cases where an error doesn't get properly propagated at the point it actually occurred. Would it be a giant pain to have this return the error if there is one?

Some other nitpicking: `HasPrefix` is a bit ambiguous, wdyt about `NextMessageIs`, or something like that?

Reply:
Done

I also checked all current uses of `HasPrefix`/`NextMessageIs` but in all cases the error can be ignored, so I did not add any extra error handling.